nncf/dynamic_graph/patterns.py

```diff
@@ -21,16 +21,18 @@
 
 BN = N('batch_norm') | N('batch_norm3d')
 
-ANY_BN_RELU_COMBO = BN + RELU | RELU + BN | BN | RELU
-
 POOLING = N('adaptive_avg_pool2d') | N('adaptive_avg_pool3d') | N('avg_pool2d') | N('avg_pool3d')
 
 NON_RELU_ACTIVATIONS = N('elu') | N('elu_') | N('prelu') | N('sigmoid') | N('gelu')
 
-SINGLE_OPS = NON_RELU_ACTIVATIONS | POOLING | N('mean') | N('layer_norm')
+ACTIVATIONS = RELU | NON_RELU_ACTIVATIONS
+
+ANY_BN_ACT_COMBO = BN + ACTIVATIONS | ACTIVATIONS + BN | BN | ACTIVATIONS
+
+SINGLE_OPS = ACTIVATIONS | POOLING | N('mean') | N('layer_norm')
```

Comment:
In this case, the ACTIVATIONS pattern enters twice into the pattern which is used for finding insertion points of quantization operations in the model. There is the ACTIVATIONS pattern in ANY_BN_ACT_COMBO and SINGLE_OPS patterns.

```
pattern = p.LINEAR_OPS | p.ARITHMETIC | p.ANY_BN_ACT_COMBO | p.LINEAR_OPS + p.ANY_BN_ACT_COMBO | p.ARITHMETIC + p.ANY_BN_ACT_COMBO | p.SINGLE_OPS | p.MATMUL 
```

Reply:
Well, it doesn't break anything - can only potentially lead to a slightly increased computational time spent pattern-matching, which is a one-time activity anyway.